extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -355,6 +356,14 @@ public void kubernetes(NonApplicationRootPathBuildItem nonApplicationRootPathBui
                                 healthConfig.startupPath)));
     }
 
+    @BuildStep
+    AdditionalBeanBuildItem shutdownHealthCheck() {
+        return AdditionalBeanBuildItem.builder()
```

Comment:
I don't know if it's possible (should be, this build step should be able to consume `ShutdownConfig`), but maybe we could do this only if a delay is configured? If it isn't, it makes no sense to have this health check, does it?

Reply:
@Ladicek exactly. The problem is that this should be a runtime configuration. The delay highly depends on the infrastructure. For example, if we configure k8s/openshift to check readiness every 10 seconds, and we configure that 3 consecutive fails of the check to constitute "un-readiness", we would need to configure a timeout of at least 40 seconds (3 * 10 for the three checks, and an additional 10 if we "just missed" the previous check).

If we have a way to create a bean only if some runtime-constaints are met, I would be glad to include the bean condietionally.

Reply:
Oh, it's a runtime config. My bad, I assumed it's a build-time config.

Reply:
And no, there's no way to include / exclude a bean based on runtime conditions.